hyperdome_client/hyperdome_client.py

```diff
@@ -279,70 +258,78 @@ def session(self):
-                                        "username": self.server.username,
-                                        "password": self.server.password})
-                self.session.post(f"{self.server.url}/therapist_signin",
-                                  data={"username": self.server.username,
-                                        "password": self.server.password})
-
-            else:
-                self.get_uid()
-                self.therapist = self.session.post(
-                    f"{self.server.url}/request_therapist",
-                    data={"guest_id": self.uid}).text
-                if self.therapist:
-                    self.is_connected = True
-        except Exception as e:
-            print(
-                ''.join(
-                    traceback.format_exception(
-                        type(e),
-                        e,
-                        e.__traceback__)))
-            Alert(
-                self.common,
-                "therapy machine broke",
-                QtWidgets.QMessageBox.Warning,
-                buttons=QtWidgets.QMessageBox.Ok)
+        if self.is_connected:
+            self.disconnect_chat()
+        if self.server_dropdown.currentIndex() == \
+           self.server_dropdown.count() - 1:
+            self.server_dropdown.setCurrentIndex(0)
+            self.start_chat_button.setEnabled(False)
+            self.server_add_dialog.exec_()
+        elif self.server_dropdown.currentIndex() != 0:
+            self.server = self.servers[self.server_dropdown.currentText()]
+            self.get_uid()
+
+    def start_chat(self):
+        @QtCore.pyqtSlot(str)
+        def after_start(counselor: str):
+            if not self.server.is_counselor and not counselor:
+                self.handle_error("No counselors available.")
+                self.start_chat_button.setEnabled(True)
+                return
+            if self.server.is_counselor:
+                self.uid = counselor
+            self.is_connected = True
+            self.get_messages_task = threads.GetMessagesTask(self.session,
+                                                             self.server,
+                                                             self.uid)
+            self.get_messages_task.setAutoDelete(False)
+            self.get_messages_task.signals.success.connect(
+                self.on_history_added)
+            self.get_messages_task.signals.error.connect(lambda: 0)
+            self.timer.start()
+            self.start_chat_button.setText("Disconnect")  # locale
+            self.start_chat_button.clicked.disconnect(
+                self.start_chat_button_connection)
+            self.start_chat_button_connection = self.start_chat_button.clicked.connect(
+                self.disconnect_chat)
+            self.start_chat_button.setEnabled(True)
+
+        self.start_chat_button.setEnabled(False)
+        start_chat_task = threads.StartChatTask(
+            self.server, self.session, self.uid)
```

Comment:
It is inconsistant. I was about to fix it, but I want to actually re-do all the QRunnable tasks relatively soon anyway because these are kludgy.

Leaving for now.